### ISSUE TITLE
Windows compile issue caused by ssize_t

### ISSUE THREAD

**Issue by shiretu** [sctplab/usrsctp#293]
Hi all,

`ssize_t` defined [here](https://github.com/sctplab/usrsctp/blob/master/usrsctplib/usrsctp.h#L75) is not guarded by anything. If the user of the lib already defined `ssize_t`, that this will result in an error of a redefined symbol.

This issue of `ssize_t` missing from windows is old and there is no fix in sight. However, here are some observations:

1. Definitions of missing things otherwise found on other platforms, like the standard ssize_t, should be define-guarded, to avoid conflicts. Yes, far from ideal, but the core problem is that all apps redefining it should make sure is 32/64 bits, depending on the target compile platform.
2. usrsctp library defines it wrongly anyways: ssize_t is not 64bit always. It depends on the target compile option. Correct approach would be along these lines (not at all authoritative, just as an example):

```
#ifndef ssize_t
	#ifdef _WIN64
		#define ssize_t __int64
	#elif defined _WIN32
		#define ssize_t int
	#else
		#error "What platform is this!?"
	#endif
#endif
```

**Comment by tuexen** [comment]
@weinrank Can you have a look at it?

**Comment by tuexen** [comment]
Fixed by #296.
